org.eclipse.xtext/src/org/eclipse/xtext/formatting2/AbstractJavaFormatter.java

```diff
@@ -0,0 +1,162 @@
+/*******************************************************************************
```

Comment:
Can we find a better name for the base class? Does it hurt to move these utils to the AbstractFormatter2?
Or maybe `BaseFormatter`? `AbstractStandardFormatter`? `AbstractFormatter3`?

Reply:
At least the reflective stuff will collide ... and the others as well, because existing formatters might use the extension method approach and then have two methods with the same signature.

I understand you want to get rid of "Java" in the name, but your suggestions seem not so nice, too. Not that I have a better idea ... BaseFormatter seems to include some basic formatting (which it don't) and AbstractFormatter3 seems not to fit (as it is the formatting2 API. If any I would go with AbstractStandardFormatter (or maybe AbstractStandardFormatter2!?)